src/bionics.cpp

```diff
@@ -639,6 +766,276 @@ void player::power_bionics()
     }
 }
 
+void player::power_bionics_new()
+{
+    // initialization
+    const int win_h = FULL_SCREEN_HEIGHT + ( TERMY - FULL_SCREEN_HEIGHT ) * 2 / 3;
+    const int win_w = FULL_SCREEN_WIDTH + ( TERMX - FULL_SCREEN_WIDTH ) / 3;
+    const int win_x = TERMX / 2 - win_w / 2;
+    const int win_y = TERMY / 2 - win_h / 2;
+
+    WINDOW *w_bionics = newwin( win_h, win_w, win_y, win_x );
+    WINDOW_PTR w_bionics_ptr( w_bionics );
+
+    WINDOW *w_bio_header = newwin( 1, win_w - 3, win_y + 1, win_x + 2 );
+    WINDOW_PTR w_bio_header_ptr( w_bio_header );
+
+    WINDOW *w_bio_description = newwin( 4, win_w - 3, win_y + win_h - 5, win_x + 2 );
+    WINDOW_PTR w_bio_description_ptr( w_bio_description );
+
+    WINDOW *w_bio_list = newwin( win_h - getmaxy( w_bio_header ) - getmaxy( w_bio_description ) -
+                                 5, win_w - 1, win_y + 4, win_x );
+    WINDOW_PTR w_bio_list_ptr( w_bio_list );
+
+    input_context ctxt( "BIONICS" );
+    ctxt.register_cardinal();
+    ctxt.register_action( "ANY_INPUT" );
+    ctxt.register_action( "REASSIGN" );
+    ctxt.register_action( "REMOVE" );
+    ctxt.register_action( "CONFIRM" );
+    ctxt.register_action( "HELP_KEYBINDINGS" );
+
+    size_t tab_count = num_bp + 2;
+    size_t tab_index = tab_count - 1;
+    int cursor = 0;
+    int scroll_position = 0;
+    int max_scroll_position;
+
+    bool recalc = true;
+    std::vector<cbm_pair> content;
+    // main loop
+    for( ;; ) {
+        if( recalc ) {
+
+            // bodyparts: all
+            if( tab_index == tab_count - 1 ) {
+                content.clear();
+                for( int bp_index = 0; bp_index <= num_bp; ++bp_index ) {
+                    std::vector<cbm_pair> more_content = define_content( g->u, static_cast<body_part>
```

Comment:
I've tried that initially but got an error. However, ```*this``` instead of ```this``` works fine, so I'll use it.